packages/@uppy/core/src/index.js

```diff
@@ -1304,11 +1311,10 @@ class Uppy {
         return this._runUpload(uploadID)
       })
       .catch((err) => {
-        const message = typeof err === 'object' ? err.message : err
-        const details = typeof err === 'object' ? err.details : null
-        this.log(`${message} ${details}`)
-        this.info({ message: message, details: details }, 'error', 4000)
-        return Promise.reject(typeof err === 'object' ? err : new Error(err))
+        if (err.isRestriction) {
+          this.emit('restriction-failed', null, err)
```

Comment:
@goto-bus-stop not too weird, `null, err`? I guess `err` should have been the first argument when this event was added originally, but that’s what we have now.

Reply:
seems ok